### PR TITLE
Add CI, tests for fmt and deny, rust-toolchain.toml

### PR DESCRIPTION
**Pull request by buffet**

### CODE REVIEW

**Comment by buffet** [comment]
Apparently the cargo-install thing actually uses some depricated API...

File: crates/bazed-rpc/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
 pub mod core_proto;
 pub mod keycode;
+
```
Comment (elkowar) [inline crates/bazed-rpc/src/lib.rs:3]: This is a weird removed line :think:

**Review by elkowar — DISMISSED** [review]
I feel like this is very work in progress

**Review by elkowar — APPROVED** [review]
Epic!

**Comment by buffet** [comment]
My first PR!!!!1

**Pull request merged**